frontend/src/main/scala/akkaviz/frontend/components/ReplTerminal.scala

```diff
@@ -11,16 +11,35 @@ import scalatags.JsDom.all._
 
 class ReplTerminal extends Component {
 
-  private def options = js.Dynamic.literal(
+  private[this] def options = js.Dynamic.literal(
     cols = 120,
     rows = 25
   ).asInstanceOf[TerminalOptions]
 
-  private var ws: js.UndefOr[WebSocket] = js.undefined
+  private[this] var ws: js.UndefOr[WebSocket] = js.undefined
 
-  private lazy val terminal = new Terminal(options)
+  private[this] lazy val terminal = new Terminal(options)
+  private[this] var connectButton: js.UndefOr[HTMLButtonElement] = js.undefined
+  private[this] var isConnected = false
 
-  private def setupWebsocket(): Unit = {
+  private[this] def connected(): Unit = {
+    isConnected = true;
+    connectButton.foreach {
+      _.innerHTML = "Disconnect"
+    }
+  }
+
+  private[this] def disconnected(): Unit = {
+    isConnected = false
+    terminal.off("data")
+    terminal.write("\n\r\n\rDisconnected\n\r\n\r")
+    ws = js.undefined
+    connectButton.foreach {
+      _.innerHTML = "Connect"
+    }
+  }
+
+  private[this] def setupWebsocket(): Unit = {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Use active names for operations with side effects](https://www.codacy.com/app/lustefaniak/akka-viz/file/2167023955/issues/source?bid=3121282&fileBranchId=3161575#l42)